src/atomex.ts

```diff
@@ -213,4 +233,90 @@ export class Atomex {
   async getSwap(swapID: string): Promise<Swap> {
     return this.makeRequest("get", `/v1/Swaps/${swapID}`, true);
   }
+
+  private getSuitableEntry(
+    entries: Entry[],
+    side: Side,
+    amount: number,
+    qtyOp: ArithmeticOP,
+    relOp: RelationalOP,
+  ): Entry {
+    let reqEntry = -1;
+    for (let i = 0; i < entries.length; i++) {
+      if (
+        entries[i].side == side &&
+        Math.max(...entries[i].qtyProfile) >= qtyOp(amount, entries[i].price)
+      ) {
+        if (reqEntry == -1) reqEntry = i;
+        else if (relOp(entries[i].price, entries[reqEntry].price)) {
+          reqEntry = i;
+        }
+      }
+    }
+    if (reqEntry == -1) {
+      throw new Error("No suitable entry found");
+    }
+    return entries[reqEntry];
+  }
+
+  getOrderPreview(
+    orderBook: OrderBook,
+    side: Side,
+    amount: number,
+    direction: "Send" | "Receive",
+  ): OrderPreview {
+    let entry: Entry, newAmount: number;
```

Comment:
This will be very hard to support.
Pls rewrite in the following way:

```
def getOrderSize(amount, price, side, direction):
  switch (side + direction) {
      case "BuySend": return amount / price;
      case "BuyReceive": return amount;
      case "SellSend": return amount;
      case "SellReceive": return amount / price;
  }

def filterEntry(entry, amount, side, direction):
  return entry.side !== side && Math.max(...entry.qtyProfile) >= getOrderSize(amount, entry.price, side, direction)

def getOrderPreview(orderBook, side, amount, direction):
  if (side == "Buy") {
    entry = orderBook.entries.find(x=> filterEntry(x, amount, side, direction))
  } else {
    entry = orderBook.entries.findLast(x=> filterEntry(x, amount, side, direction))
  }
```